### PR TITLE
Move away from CommonJS

### PR DESCRIPTION
**Pull request by nmattia**
This changes our ts and project config to use es as opposed to commonjs. This will make future changes (moving to vite, SSR support) simpler.

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
+import { config } from "dotenv";
 
-require("dotenv").config();
+const __dirname = fileURLToPath(new URL(".", import.meta.url));
+
+config();
 
 /** Read the II canister ID from dfx's local state */
-function readCanisterId() {
+async function readCanisterId() {
```
Comment (peterpeterparker) [inline webpack.config.js:17]: not sure `async` was actually ever needed, is it?

Comment (nmattia) [reply webpack.config.js:17]: Good catch, thanks

File: tsconfig.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
     "esModuleInterop": true,
     "forceConsistentCasingInFileNames": true,
     "lib": ["ES2019", "DOM"],
-    "module": "commonjs",
+    "module": "esnext",
```
Comment (peterpeterparker) [inline tsconfig.json:9]: does it works with `es2020` as well? maybe we can use it to be futurstic by using modules but be a bit preservative at the same time by using a well supported standard?

File: webpack.config.js [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
+import CopyPlugin from "copy-webpack-plugin";
+import CompressionPlugin from "compression-webpack-plugin";
+import HtmlWebpackPlugin from "html-webpack-plugin";
+import MiniCssExtractPlugin from "mini-css-extract-plugin";
+import CssMinimizerPlugin from "css-minimizer-webpack-plugin";
+import { fileURLToPath } from "url";
+import { config } from "dotenv";
 
-require("dotenv").config();
+const __dirname = fileURLToPath(new URL(".", import.meta.url));
```
Comment (peterpeterparker) [inline webpack.config.js:12]: well played 👍

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

File: webpack.config.js (hunk 3 above)
Comment (nmattia) [reply webpack.config.js:12]: 100% stackoverflow 😉

File: tsconfig.json (hunk 2 above)
Comment (nmattia) [reply tsconfig.json:9]: Done!

**Pull request merged**